### ISSUE TITLE
Consider moving Npcap (and then perhaps Nmap) installers from NSIS to MSI

### ISSUE THREAD

**Issue by fyodor** [nmap/npcap#103]
This needs further research, but we are adding this issue to track our existing knowledge on this topic and to invite anyone else to comment if they have feedback about pros/cons, implementation ideas, potential obstacles, etc.

Nmap and Npcap currently use self-executable installers generated by NSIS (http://nsis.sourceforge.net/Main_Page).  They our signed by our codesigning certificate.  This has worked well for us for more than a decade (Nmap introduced our NSIS installer in January 2006).  However, Microsoft Windows currently has a native software management system called Windows Installer, which processes MSI and/or MSP database files.  MSI files are also often distributed as part of EXE bundles that first check whether the system's MSIEXEC.EXE exists and is new enough.  If not, they unpack their own MSIEXEC and use that to install the MSI.

Since our signed NSIS EXE installer seems to be working well for both Nmap and Npcap, and has had more than a decade of debugging and improvements (in the Nmap case), we would probably only switch (or add a separate MSI option) if we find a compelling reason to.

It is worth noting that tools are avaialble for converting NSIS EXE files into MSI ones, such as http://www.exemsi.com/.  But it isn't clear whether that sort of MSI would be a big help for anyone or not.

**Comment by fyodor** [comment]
I've moved this to the new Npcap issues tracker since Npcap is the top priority for the MSI migration.  If we do that and it works well, we can consider doing Nmap

**Comment by fyodor** [comment]
For our tracking purposes I wanted to note that another licensee just requested this.  And I wanted to let everyone know that it is still high on our priority list.  We just have been working on [a lot of other great stuff in the meantime](https://npcap.org/changelog).  Our current main focus is WHQL certification.

**Comment by guyharris** [comment]
In #768 an OEM user says that "The current oem npac[sic] installer is a 32-bit executable. Some of our customers object to any 32-bit-running things on their Windows machines." I asked whether an MSI installer would work.

Note that this means that if any helper executables are needed, as per item 3 "You can still use NPFInstall.exe (or a DLL) to handle custom actions which are not offered out of the box." in https://github.com/nmap/npcap/issues/103#issuecomment-623607131, there would presumably have to be both 32-bit and 64-bit versions for customers such as the one mentioned, with the appropriate one run for the platform - at least as long as Npcap supports 32-bit machines.

**Comment by fyodor** [comment]
Good point @guyharris.  And Npcap will probably have to support 32-bit systems for many more years.  Microsoft still offers free 32-bit Windows 10 systems until next October, and then will offer paid support for years after that.  There are also fairly recent vintage ARM64 systems which supported x86 emulation but didn't yet support x64.  We could offer three separate installers (and supporting helper programs), but it's hard to justify the complexity for us and the users without some clear and convincing justification.  It's really nice to have one compatible installer (or perhaps in the future MSI package) which can then detect and install the appropriate-architecture binaries for each system.
